QEfficient/generation/text_generation_inference.py

```diff
@@ -235,111 +272,102 @@ def cloud_ai_100_exec_kv_helper(
     enable_debug_logs: bool = False,
     stream: bool = True,
     write_io_dir: Optional[str] = None,
     automation=False,
 ):
-    if batch_size == 1:
-        prefill_time = []
-        decode_perf = []
-        total_perf = []
-        total_time = []
-        generated_texts = []
-        for i in range(len(prompt)):
-            latency_stats = cloud_ai_100_exec_kv_helper(
-                tokenizer=tokenizer,
-                prompt=[prompt[i]],
-                qpc=qpc_path,
-                device_id=device_id,
-                ctx_len=ctx_len,
-                generation_len=generation_len,
-                enable_debug_logs=enable_debug_logs,
-                stream=stream,
-                write_io_dir=write_io_dir,
-            )
-            generated_texts.append(latency_stats[0])
-            prefill_time.append(latency_stats[1])
-            decode_perf.append(latency_stats[2])
-            total_perf.append(latency_stats[3])
-            total_time.append(latency_stats[4])
-
-        prefill_time = np.average(prefill_time)
-        decode_perf = np.average(decode_perf)
-        total_perf = np.average(total_perf)
-        total_time = np.average(total_time)
-
-    else:
-        latency_stats = cloud_ai_100_exec_kv_helper(
+    batch_size, ctx_len = get_compilation_dims(qpc_path)
+    prompt: List[str] = get_input_prompts(prompt, prompts_txt_file_path)
+    prompt = fix_prompts(prompt, batch_size)
+
+    prefill_time = []
+    decode_perf = []
+    total_perf = []
+    total_time = []
+    generated_texts = []
+    generated_ids = []
+
+    for i in range(0, len(prompt), batch_size):
+        execinfo = cloud_ai_100_exec_kv_helper(
             tokenizer=tokenizer,
-            prompt=prompt,
-            qpc=qpc_path,
+            prompt=prompt[i : i + batch_size],
+            qpc_path=qpc_path,
             device_id=device_id,
             ctx_len=ctx_len,
             generation_len=generation_len,
             enable_debug_logs=enable_debug_logs,
             stream=stream,
             write_io_dir=write_io_dir,
         )
-        generated_texts, prefill_time, decode_perf, total_perf, total_time = latency_stats
-
+        generated_ids.append(execinfo.generated_ids)
+        generated_texts.append(execinfo.generated_texts)
+        prefill_time.append(execinfo.prefill_time)
+        decode_perf.append(execinfo.decode_perf)
+        total_perf.append(execinfo.total_perf)
+        total_time.append(execinfo.total_time)
+
+    prefill_time = np.average(prefill_time)
+    decode_perf = np.average(decode_perf)
+    total_perf = np.average(total_perf)
+    total_time = np.average(total_time)
+
+    execinfo = CloudAI100ExecInfo(
+        batch_size=batch_size,
+        generated_texts=generated_texts,
+        generated_ids=generated_ids,
+        prefill_time=prefill_time,
+        decode_perf=decode_perf,
+        total_perf=total_perf,
+        total_time=total_time,
+    )
     print_latency_stats_kv(
         prompt,
```

Comment:
This is (still) printing to stdout, rather than just returning.  Could we just return at this layer, so we have a clean API to call programmatically?

Reply:
Hi @carlstreeter-quic , we want to support logger for printing/saving such info in log fiiles.
So, user can setup logger as they want and handle this info, it won't `print` by default.
We plan to do this as follow up to this PR.
This PR has already grown a lot.